### ISSUE TITLE
samples/net/test_15_4 cannot be built by sanitycheck

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#618]
**_Reported by Andrew Boie:_**

This test case didn't have a testcase.ini so it was never built, which is probably why this wasn't reported earlier.

According to the README, I should be able to do "make qemu0" to do a loopback test, but the build simply fails:

```
$ make qemu0
make[1]: Entering directory '/projects/zephyr3'
make[2]: Entering directory '/projects/zephyr3/samples/net/test_15_4/outdir'
  Using /projects/zephyr3 as source for kernel
  GEN     ./Makefile
  CHK     include/generated/version.h
  CHK     misc/generated/configs.c
  CHK     include/generated/offsets.h
  CC      drivers/ieee802154/ieee802154_cc2520.o
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c: In function 'set_reset':
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:298:31: error: 'CC2520_GPIO_IDX_RESET' undeclared (first use in this function)
  gpio_pin_write(cc2520->gpios[CC2520_GPIO_IDX_RESET],
                               ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:298:31: note: each undeclared identifier is reported only once for each function it appears in
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:299:10: error: 'CONFIG_CC2520_GPIO_RESET' undeclared (first use in this function)
          CONFIG_CC2520_GPIO_RESET, value);
          ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c: In function 'set_vreg_en':
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:306:31: error: 'CC2520_GPIO_IDX_VREG_EN' undeclared (first use in this function)
  gpio_pin_write(cc2520->gpios[CC2520_GPIO_IDX_VREG_EN],
                               ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:307:10: error: 'CONFIG_CC2520_GPIO_VREG_EN' undeclared (first use in this function)
          CONFIG_CC2520_GPIO_VREG_EN, value);
          ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c: In function 'get_fifo':
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:314:30: error: 'CC2520_GPIO_IDX_FIFO' undeclared (first use in this function)
  gpio_pin_read(cc2520->gpios[CC2520_GPIO_IDX_FIFO],
                              ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:315:9: error: 'CONFIG_CC2520_GPIO_FIFO' undeclared (first use in this function)
         CONFIG_CC2520_GPIO_FIFO, &pin_value);
         ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c: In function 'get_fifop':
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:324:30: error: 'CC2520_GPIO_IDX_FIFOP' undeclared (first use in this function)
  gpio_pin_read(cc2520->gpios[CC2520_GPIO_IDX_FIFOP],
                              ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:325:9: error: 'CONFIG_CC2520_GPIO_FIFOP' undeclared (first use in this function)
         CONFIG_CC2520_GPIO_FIFOP, &pin_value);
         ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c: In function 'get_cca':
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:334:30: error: 'CC2520_GPIO_IDX_CCA' undeclared (first use in this function)
  gpio_pin_read(cc2520->gpios[CC2520_GPIO_IDX_CCA],
                              ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:335:9: error: 'CONFIG_CC2520_GPIO_CCA' undeclared (first use in this function)
         CONFIG_CC2520_GPIO_CCA, &pin_value);
         ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c: In function 'enable_fifop_interrupt':
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:374:42: error: 'CC2520_GPIO_IDX_FIFOP' undeclared (first use in this function)
   gpio_pin_enable_callback(cc2520->gpios[CC2520_GPIO_IDX_FIFOP],
                                          ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:375:7: error: 'CONFIG_CC2520_GPIO_FIFOP' undeclared (first use in this function)
       CONFIG_CC2520_GPIO_FIFOP);
       ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c: In function 'enable_sfd_interrupt':
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:386:42: error: 'CC2520_GPIO_IDX_SFD' undeclared (first use in this function)
   gpio_pin_enable_callback(cc2520->gpios[CC2520_GPIO_IDX_SFD],
                                          ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:387:7: error: 'CONFIG_CC2520_GPIO_SFD' undeclared (first use in this function)
       CONFIG_CC2520_GPIO_SFD);
       ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c: In function 'setup_gpio_callbacks':
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:399:33: error: 'CONFIG_CC2520_GPIO_SFD' undeclared (first use in this function)
       sfd_int_handler, BIT(CONFIG_CC2520_GPIO_SFD));
                                 ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:400:34: error: 'CC2520_GPIO_IDX_SFD' undeclared (first use in this function)
  gpio_add_callback(cc2520->gpios[CC2520_GPIO_IDX_SFD],
                                  ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:404:35: error: 'CONFIG_CC2520_GPIO_FIFOP' undeclared (first use in this function)
       fifop_int_handler, BIT(CONFIG_CC2520_GPIO_FIFOP));
                                   ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:405:34: error: 'CC2520_GPIO_IDX_FIFOP' undeclared (first use in this function)
  gpio_add_callback(cc2520->gpios[CC2520_GPIO_IDX_FIFOP],
                                  ^
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c: At top level:
/projects/zephyr3/drivers/ieee802154/ieee802154_cc2520.c:772:12: warning: 'cc2520_set_txpower' defined but not used [-Wunused-function]
 static int cc2520_set_txpower(struct device *dev, short dbm)
            ^
/projects/zephyr3/scripts/Makefile.build:182: recipe for target 'drivers/ieee802154/ieee802154_cc2520.o' failed
make[4]: *** [drivers/ieee802154/ieee802154_cc2520.o] Error 1
/projects/zephyr3/scripts/Makefile.build:356: recipe for target 'drivers/ieee802154' failed
make[3]: *** [drivers/ieee802154] Error 2
/projects/zephyr3/Makefile:896: recipe for target 'drivers' failed
make[2]: *** [drivers] Error 2
make[2]: Leaving directory '/projects/zephyr3/samples/net/test_15_4/outdir'
Makefile:175: recipe for target 'sub-make' failed
make[1]: *** [sub-make] Error 2
make[1]: Leaving directory '/projects/zephyr3'
/projects/zephyr3/samples/net/test_15_4/Makefile.15_4:74: recipe for target 'qemu0' failed
make: *** [qemu0] Error 2
```

Please fix this build error and then add a testcase.ini so that CI at least builds this application.

(Imported from Jira ZEP-697)

**Comment by nashif** [comment]
**_by Jaakko Hannikainen:_**

Fixed with https://gerrit.zephyrproject.org/r/4202

**Comment by nashif** [comment]
**_by Andrew Boie:_**

Please don't resolve tickets until patches have been merged into the master branch.

**Comment by nashif** [comment]
**_by Jaakko Hannikainen:_**

Yes, my bad, sorry...

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Hi _Andrew Boie_ , as the Reporter of this bug, can you please verify it's fixed?  It looks like it just got re-resolved yesterday.  Thanks!

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Correcting the priority field
